shakenfist/blob.py

```diff
@@ -137,6 +166,32 @@ def observe(self):
                 info['mime-type'] = magic.Magic(mime=True).from_file(blob_path)
                 self._db_set_attribute('info', info)
 
+    def ref_count_inc(self):
+        with self.get_lock_attr('ref_count', 'Increase ref count'):
+            if self.state.value == self.STATE_DELETED:
+                raise BlobDeleted
+            new_count = self.ref_count + 1
+            self._db_set_attribute('ref_count', {'ref_count': new_count})
+            return new_count
+
+    def ref_count_dec(self):
+        with self.get_lock_attr('ref_count', 'Increase ref count'):
+            new_count = self.ref_count - 1
+            if new_count < 0:
+                raise BlobRefCountDecrementBelowZero
```

Comment:
I'm actually not sure if we should fail here. I guess we should so we notice a bug, but its weird that my instance delete would fail entirely because of a reference counting error.

Reply:
The exception would be alerting that a major bug has occurred.

The only way to get a negative value is to start at zero. If the counter is at zero then the state will (should) be DELETED. If not state deleted, yet another bug has been uncovered.

Reply:
Fair.